test/CMakeLists.txt

```diff
@@ -14,30 +14,36 @@
 
 cmake_minimum_required(VERSION 3.0 FATAL_ERROR)
 
-include(CTest)
-
-set(CURRENT_TEST_BINARY_DIR ${CMAKE_BINARY_DIR}/testBin/)
-
-file(GLOB_RECURSE TEST_SRCS "*.c")
-
-foreach(case_file ${TEST_SRCS})
+macro(add_test_case case_file)
   get_filename_component(case_name ${case_file} NAME_WE)
 
   add_executable(${case_name} ${case_file})
-  target_link_libraries(${case_name}
-    PRIVATE xaptumtpm-static
-    )
+
+  if(BUILD_SHARED_LIBS)
+    target_link_libraries(${case_name} PRIVATE xaptum-tpm)
+  else()
+    target_link_libraries(${case_name} PRIVATE xaptum-tpm_static)
```

Comment:
Should this be `PRIVATE xaptum-tpm${STATIC_SUFFIX}`?

Reply:
Ah, nevermind. I misunderstood cmake target_link_libraries.

Ignore this comment